scripts/lumen_cli/cli/lib/common/file_utils.py

```diff
@@ -79,3 +136,35 @@ def read_yaml_file(file_path: str) -> dict:
         raise RuntimeError(
             f"Unexpected error while reading YAML file '{file_path}': {e}"
         ) from e
+
+
+def local_image_exists(image_name: str):
+    """
+    Check if a local Docker image exists.
+    image name format: <image_name>:<tag>
+    """
+    try:
+        run_cmd(f"docker image inspect {image_name}", log_cmd=False)
+        return True
+    except subprocess.CalledProcessError:
+        return False
+
+
+def clone_external_repo(target: str, repo: str, cwd: str):
+    logger.info(f"cloning {target}....")
+    commit = get_post_build_pinned_commit(target)
+
+    # delete the directory if it exists
+    remove_dir(cwd)
+
+    # Clone the repo & checkout commit
+    run_cmd(f"git clone {repo}")
```

Comment:
Same comment on the wide spread use of `run_cmd`, i think you could consider adding those functionality to `utils.py`:

1. Handle Git operations with https://pypi.org/project/GitPython
2. Handle Docker operations with https://pypi.org/project/docker
3. File operations with `shutil`

The CLI will looks much better with them IMO